packages/datadog-plugin-cassandra-driver/src/index.js

```diff
@@ -3,8 +3,16 @@
 const tx = require('../../dd-trace/src/plugins/util/tx')
 
 function createWrapInnerExecute (tracer, config) {
+  const isValid = (args) => {
+    return args.length === 4 || typeof args[3] === 'function'
+  }
+
   return function wrapInnerExecute (_innerExecute) {
     return function _innerExecuteWithTrace (query, params, execOptions, callback) {
+      if (!isValid(arguments)) {
```

Comment:
should we track a metric here, or maybe unwrap?

this happens if the API changes right? if we are able to back out/unwrap when we notice the API change, could avoid the overhead of our function entirely

Reply:
We have no guarantee that it would fail 100% of the time, and some traces is better than no traces. It's also possible that only part of the integration is not working properly, but we are still able to get some level of tracing. It's all about trade-offs. We can always revisit later, but for now I would say let's focus on reducing the likelihood of such change blowing up.